kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -102,8 +98,9 @@ public TopicPartitionWriter(TopicPartition tp,
 
     flushSize = connectorConfig.getInt(S3SinkConnectorConfig.FLUSH_SIZE_CONFIG);
     topicsDir = connectorConfig.getString(StorageCommonConfig.TOPICS_DIR_CONFIG);
-    rotateIntervalMs = connectorConfig.getLong(S3SinkConnectorConfig.ROTATE_INTERVAL_MS_CONFIG);
-    rotateScheduleIntervalMs = connectorConfig.getLong(S3SinkConnectorConfig.ROTATE_SCHEDULE_INTERVAL_MS_CONFIG);
+    rotateIntervalMs = partitioner instanceof TimeBasedPartitioner ?
```

Comment:
I was thinking that using a single interval (duration for that matter) to adjust the granularity of time buckets and then depend on file size for additional file rotation within a time bucket would be good enough. Giving it a second thought, I agree that treating time buckets as more static, and using a time-based configurable interval, such as `rotate.interval.ms`, besides flush size to rotate files within the bucket makes sense. Will allow for more fine-grain rotation. 

Having said that, this property doesn't solve the issue of outstanding messages. Since we don't use scheduled intervals (wallclock based) we always depend on the "next" message to perform rotation, file-based or time-based. For instance, with hourly time-buckets, and rotation interval of 20 minutes, if messages suddenly stop at, let's say, hh:58 then all records between hh:40 and hh:59 will remain outstanding until the "next record" arrives.

I feel that's a reasonable trade-off for exactly-once time-based partitioning.